generator/lib/src/realm_field_info.dart

```diff
@@ -80,8 +80,13 @@ class RealmFieldInfo {
   Iterable<String> toCode() sync* {
     final getTypeName = type.isRealmCollection ? basicMappedTypeName : basicNonNullableMappedTypeName;
     yield '@override';
-    final validateBacklink = isRealmBacklink ? ", validateBacklink: true" : "";
-    yield "$mappedTypeName get $name => RealmObjectBase.get<$getTypeName>(this, '$realmName'$validateBacklink) as $mappedTypeName;";
+    if (isRealmBacklink) {
+      yield "$mappedTypeName get $name {";
+      yield "if(!isManaged){throw RealmException('Using backlinks is only possible for managed(persisted) objects.');}";
```

Comment:
I would suggest formatting this a bit here. The generator code should be readable by us

Reply:
no need to use persisted when we say managed and this is the word we are using across the docs and api docs imo
```suggestion
      yield "if(!isManaged){throw RealmException('Using backlinks is only possible for managed objects.');}";
```